include/mneme/displs.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef MNEME_DISPLS_H
+#define MNEME_DISPLS_H
+
+#include <cassert>
+#include <cstddef>
+#include <iterator>
+#include <utility>
+#include <vector>
+
+namespace mneme {
+
+template <typename IntT> class DisplsIterator {
+public:
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = std::pair<std::size_t, std::size_t>;
+    using pointer = value_type*;
+    using reference = value_type&;
+
+    DisplsIterator(std::vector<IntT> const& displs, bool end = false) : displs(displs), p(0), i(0) {
+        if (end) {
+            p = displs.size() - 1;
+            i = displs[p];
+        } else {
+            i = displs[0];
+            next();
+        }
+    }
+
+    bool operator!=(DisplsIterator const& other) {
+        return i != other.i || p != other.p || &displs != &other.displs;
+    }
+    bool operator==(DisplsIterator const& other) { return !(*this != other); }
```

Comment:
Is there a reason why this isnt *this == other?

Reply:
?